Api/V8/JsonApi/Response/DocumentResponse.php

```diff
@@ -0,0 +1,86 @@
+<?php
+namespace Api\V8\JsonApi\Response;
+
+class DocumentResponse implements \JsonSerializable
+{
+    /**
+     * @var DataResponse|DataResponse[]|[]
+     */
+    private $data;
+
+    /**
+     * @var MetaResponse
+     */
+    private $meta;
+
+    /**
+     * @var LinksResponse
+     */
+    private $links;
+
+    /**
+     * @return DataResponse|DataResponse[]|[]
+     */
+    public function getData()
+    {
+        return $this->data;
+    }
+
+    /**
+     * @param DataResponse|DataResponse[]|[] $data
+     */
+    public function setData($data)
+    {
+        $this->data = $data;
+    }
+
+    /**
+     * @return MetaResponse
+     */
+    public function getMeta()
+    {
+        return $this->meta;
+    }
+
+    /**
+     * @param MetaResponse $meta
+     */
+    public function setMeta(MetaResponse $meta)
+    {
+        $this->meta = $meta;
+    }
+
+    /**
+     * @return LinksResponse
+     */
+    public function getLinks()
+    {
+        return $this->links;
+    }
+
+    /**
+     * @param LinksResponse $links
+     */
+    public function setLinks(LinksResponse $links)
+    {
+        $this->links = $links;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function jsonSerialize()
+    {
+        if (!$this->getData()) {
+            $this->setMeta(new MetaResponse(['message' => 'Request was successful, but there is no result']));
+        }
+
+        $response = [
+            'meta' => $this->getMeta(),
+            'data' => $this->getData(),
+            'links' => $this->getLinks(),
+        ];
+
+        return array_filter($response);
```

Comment:
@zoltankocsardi 
If I'm understanding the JSON Api correctly, I believe that this allows an invalid response. 
If you query a module with no records in it (say, Alerts), then an empty array is returned. 

According to the [spec](http://jsonapi.org/format/#document-top-level), we have to return _something_. I think the most logical thing would be to return the "data" key with the value being a blank array.

```
{
    data : []
}
```

Reply:
Another nice catch. I thought data must contain id and type [otherwise just show a message in the meta object like there is no result], but yes, it could be null or an empty array in case of collections. Gonna refine this as well. Thanks again! :) :beer: